Sources/RxAlamofire.swift

```diff
@@ -985,11 +987,13 @@ extension Reactive where Base: DataRequest {
                     observer.on(.completed)
                 }
             }
-            
-            if let upload = self.base as? UploadRequest {
-                upload.uploadProgress(closure: handler)
-            } else {
-                self.base.downloadProgress(closure: handler)
+
```

Comment:
@sunshinejr From your Alamofire experience, does this conversion table make sense to you ?

Reply:
Yeah I think it's correct, the only thing is that `upload` should be before `data` because `UploadRequest` inherits from `DataRequest`, but you're already did it so it's all good. Maybe we could add a comment about it?

Reply:
About the priority order? Sure, I'll add that.

Reply:
Added [here](https://github.com/RxSwiftCommunity/RxAlamofire/pull/111/commits/8d2fa4f50f436e4ef6b97be54c98677c69b37c4e) LMK if makes sense to you :)